tiled/client/array.py

```diff
@@ -167,15 +171,51 @@ def write(self, array):
             )
         )
 
-    def write_block(self, array, block):
+    def write_block(self, array, block, slice=...):
         handle_error(
             self.context.http_client.put(
                 self.item["links"]["block"].format(*block),
                 content=array.tobytes(),
                 headers={"Content-Type": "application/octet-stream"},
+                params=params_from_slice(slice),
             )
         )
 
+    def patch(self, array: NDArray, slice: NDSlice, extend=False):
+        """
+        Write data
+
+        Parameters
+        ----------
+        array : array-like
+            The data to write
+        slice : NDSlice
+            Where to place this data in the array
```

Comment:
I think it would be nice to give more information on how to format the slice parameter.